Runtime/AvatarCreator/UI/Elements/TemplateSelectionElement.cs

```diff
@@ -26,55 +27,27 @@ private void Awake()
         public async void LoadAndCreateButtons()
         {
             await LoadTemplateData();
-            CreateButtons();
-            await LoadTemplateRenders();
+            CreateButtons(avatarTemplates.ToArray(), async (button, asset) =>
+            {
+                var downloadHandler = new DownloadHandlerTexture();
```

Comment:
We really need to wrap this into some sort of reusable image download request class. I know it was mentioned we had one somewhere but I couldn't find it 😅